sherpa/ui/tests/test_plot_pvalue.py

```diff
@@ -69,11 +64,326 @@ def test_plot_pvalue(make_data_path, clean_astro_ui, hide_log_output):
+
+    # Check the plot
+    #
+    tmp = ui.get_pvalue_plot()
+
+    assert tmp.lr == pytest.approx(2.679487496941789)
+
+    assert tmp.xlabel == 'Likelihood Ratio'
+    assert tmp.ylabel == 'Frequency'
+    assert tmp.title == 'Likelihood Ratio Distribution'
+
+    # It would be nice to check the values here
+    #
+    assert tmp.ratios.shape == (100, )
+    assert tmp.xlo.shape == (21, )
+    assert tmp.xhi.shape == (21, )
+    assert tmp.y.shape == (21, )
+
+
+@pytest.fixture
+def setup_imgdata_model():
+    """Use a model for the PSF"""
+
+    # Fake an image
+    #
+    x1, x0 = np.mgrid[0:8, 0:10]
+    ymod = 10 + 100 / ((x0 - 5.5)**2 + (x1 - 3.5)**2)
+
+    x0 = x0.flatten()
+    x1 = x1.flatten()
+    y = ymod.flatten()
+
+    y = y.astype(int)  # convert to integers
+
+    ui.load_arrays(1, x0, x1, y, (8, 10), ui.DataIMG)
+
+    pmodel = ui.create_model_component('gauss2d', 'pmodel')
+    pmodel.xpos = 0
+    pmodel.ypos = 0
+    pmodel.fwhm = 3
+
+    ui.load_psf('psf', pmodel)
+    psf.size = [10, 10]  # not sure if this is useful but leave in
+    psf.center = [5, 4]
+    ui.set_psf(psf)
+
+    setup_imgdata_source()
+
+
+def setup_imgdata_source():
+    """Set the model and fit setup"""
+    ui.set_source(ui.gauss2d.g1 + ui.const2d.c1)
+
+    c1.c0 = 10
+    g1.fwhm = 4
+    g1.xpos = 4
+    g1.ypos = 3
+    g1.ampl = 100
+
+    ui.set_stat('cstat')
+    ui.set_method("simplex")
+
+    # Fit and check we are in the expected place (so this is a
+    # pre-condition for the tests).
+    #
+    ui.fit()
+    assert ui.calc_stat() == pytest.approx(362.9264257040166)
+
+
+def fake_psf_image(outfile):
+    """Create a PSF image
+
+    This is based on the output of image_psf() saved to a file
+    for the test_plot_pvalue_imgpsf_convolved case. As this is
+    a very-small file and it could change (to make the test more
+    meaningful) we do not add the data to the repository but
+    create it directly.
+
+    """
+
+    hdr = ['SIMPLE  =                    T /',
+           'BITPIX  =                  -64 / Bits per pixel',
+           'NAXIS   =                    2 /',
+           'NAXIS1  =                   10 /',
+           'NAXIS2  =                    8 /']
+
+    x1, x0 = np.mgrid[0:8, 0:10]
+    dx0 = (x0 - 5).flatten()
+    dx1 = (x1 - 4).flatten()
+
+    mdl = Gauss2D()
+    mdl.fwhm = 3
+    y = mdl(dx0, dx1)
+    y = y / y.max()
+
+    # Create the FITS structure
+    #
+    with open(outfile, "wb") as fh:
+        for h in hdr:
+            fh.write(f'{h:80s}'.encode('ascii'))
+
+        n = 2880 - (len(hdr) + 1) * 80
+
+        # assume we don't have enough lines to pass 1 block!
+        assert n > 0
+
+        fmt = f'{{:{n}s}}'
+        fh.write(fmt.format(' ').encode('ascii'))
+
+        fh.write('{:80s}'.format('END').encode('ascii'))
+
+        # FITS is big-endian
+        yb = y.astype('>f8').tobytes()
+        fh.write(yb)
+
+        n = 2880 - len(yb)
+        assert n > 0
+
+        fh.write(b'\0' * n)
```

Comment:
I somehow have a dislike to writing fits files "by hand", but well...

Reply:
I was tempted to add some utility functions so at least the logic is in one place and can be vetted. The advantage is that we are then not tied to using the I/O backend, which would require more work extending the backend API **and** you are then stuck with testing whatever they create rather than a possibly-imperfect file that you can create manually. An alternative is to just add such a small file to the test directory but at the moment this is just acting as a regression test and I could see someone replacing it with a better image if we ever get around to improving the testing of the pvalue stuff.

Reply:
Oh I was not asking for a change - just commenting.